docs/learn/expert-guides/deploy-multichain-contracts.md

```diff
@@ -0,0 +1,594 @@
+// contract instances and bytecode available from the last step
+
+// Precompute the address for ERC1167 proxy deployment
+const precomputedProxyAddress =
+  await lsp16UniversalFactory.computeERC1167Address(
+    precomputedAddressWithInit,
+    deploymentSalt,
+    false, // --> boolean indicating whether the contract is initializable
+    '0x', // --> bytes representing the initialize data
+  );
+
+// Deploy ERC1167 proxy
+const deployProxyTx = await lsp16UniversalFactory.deployERC1167Proxy(
+  precomputedAddressWithInit,
+  deploymentSalt,
+);
+
+await deployProxyTx.wait();
+
+const proxyContract = new ethers.Contract(
+  precomputedProxyAddress,
+  TargetContractABI,
+  signer,
+);
+
+const numberInProxyWithoutInit = await proxyContract.number();
+// The number will be 0, as a proxy does not have state unless its initialized
+console.log(
+  'The number in proxy without initialization is: ',
+  numberInProxyWithoutInit,
+);
+```
+
+### Case 4: ERC-1167 Proxy contract with initializable function
+
+Using the `LSP16UniversalFactory`, we can also deploy initializable minimal proxies according to the ERC1167 standard, where a method can be called to generate a minimal proxy based on the address of an existing contract and have a call after deployment to initialize the proxy.
+
+```js title="main.js"
+// Encode function call for proxy initialization
+const encodedFunctionCallForProxyInit =
+  proxyContract.interface.encodeFunctionData('setNumber', [789]); // Dummy value
+
+// Precompute the address for initialized ERC1167 proxy
+const precomputedInitializedProxyAddress =
+  await lsp16UniversalFactory.computeERC1167Address(
+    precomputedAddressWithInit,
+    deploymentSalt,
+    true, // --> boolean indicating whether the contract is initializable
+    encodedFunctionCallForProxyInit, // --> bytes representing the initialize data
+  );
+
+// Deploy and initialize ERC1167 proxy
+const deployAndInitProxyTx =
+  await lsp16UniversalFactory.deployERC1167ProxyAndInitialize(
+    precomputedAddressWithInit,
+    deploymentSalt,
+    encodedFunctionCallForProxyInit,
+  );
+await deployAndInitProxyTx.wait();
+
+const initializedProxyContract = new ethers.Contract(
+  precomputedInitializedProxyAddress,
+  TargetContractABI,
+  signer,
+);
+const numberInProxyAfterInit = await initializedProxyContract.number();
+console.log('The number in the initialized proxy is: ', numberInProxyAfterInit);
+```
+
+### Generating CREATE2 Salt
+
+This section is primarily intended for advanced developers who require knowledge of the specific salt used in address generation. Developers focusing on basic deployment across multiple chains may not need this detailed information and can precompute the address of the contract to be created using the provided method of the contract.
+
+The `LSP16UniversalFactory` uses a unique approach to deploying contracts with CREATE2. Instead of using the salt directly, it combines and hashes the salt with other parameters.
```

Comment:
```suggestion
The `LSP16UniversalFactory` uses a unique approach to deploy contracts with CREATE2. Instead of using the salt directly, it combines and hashes the salt with other parameters.
```